assets/js/translator.js

```diff
@@ -1252,6 +1249,52 @@ function autoSelectDstLang() {
     $('#dstLangSelect').val(curDstLang).change();
 }
 
+function setDefaultSrcLang() {
+    // Default for new users is first available Browser Preference Language pair
+    var browserLangs = window.navigator.languages; // Chrome, Mozilla and Safari
+    var ieLang = window.navigator.userlanguage || window.navigator.browserlanguage; // Internet Explorer
+    var prefSrcLang;
```

Comment:
I don't think we're preserving the behavior anymore where a user who comes back to the site gets the same source language they switched to. Are we?

Reply:
Yeahh... i didnt notice it. I can Fix it. :)

Reply:
Done. :)

Reply:
I don't see how? `curSrcLang = iso639CodesInverse[prefSrcLang];` will happen with `prefSrcLang` not set.

Reply:
Where?

Reply:
'if(!prefSrcLang) {
        ....
    }
    else {
        curSrcLang = iso639CodesInverse[prefSrcLang];
        curDstLang = pairs[curSrcLang][0];
    }'

Reply:
its happen after prefSrcLang is set

Reply:
ah, right